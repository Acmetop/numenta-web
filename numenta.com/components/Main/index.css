/* Numenta company website source code. Copyright Â© 2016 Numenta.*/
/* Full details in LICENSE.txt, or contact us at <http://numenta.com>.*/
/* This program is free software: you can redistribute it and/or modify it under*/
/* the terms of the GNU Affero General Public License as published by the Free*/
/* Software Foundation, either version 3 of the License, or (at your option) any*/
/* later version. This program is distributed in the hope that it will be*/
/* useful, but WITHOUT ANY WARRANTY; without even the implied warranty of*/
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero*/
/* General Public License for more details. You should have received a copy of*/
/* the GNU Affero General Public License along with this program. If not, see*/
/* <https://www.gnu.org/licenses/agpl.html>.*/

.main {
  composes: bg-white from 'tachyons';
  composes: br--bottom from 'tachyons';
  composes: br2 from 'tachyons';
  composes: center from 'tachyons';
  composes: mt5 from 'tachyons';
  composes: mw-100 from 'tachyons';
  composes: mw7-m from 'tachyons';
  composes: mw8-l from 'tachyons';
  composes: overflow-hidden from 'tachyons';
  composes: pb4 from 'tachyons';
  composes: ph3 from 'tachyons';
  composes: ph4-l from 'tachyons';
  composes: pt3 from 'tachyons';
  composes: shadow-4 from 'tachyons';
}
