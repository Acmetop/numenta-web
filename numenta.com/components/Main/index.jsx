import React from 'react'

import styles from './style.css'


const Main = ({children}) => (
  <main className={styles.main}>
    {children}
  </main>
)

Main.propTypes = {
  children: React.PropTypes.any.isRequired,
}

export default Main
