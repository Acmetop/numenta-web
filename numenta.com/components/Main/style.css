.main {
  composes: bg-white from 'tachyons';
  composes: center from 'tachyons';
  composes: mt5 from 'tachyons';
  composes: mw-100 from 'tachyons';
  composes: mw7-m from 'tachyons';
  composes: mw8-l from 'tachyons';
  composes: ph3 from 'tachyons';
  composes: ph4-l from 'tachyons';
  composes: pb4 from 'tachyons';
  composes: pt3 from 'tachyons';
}
