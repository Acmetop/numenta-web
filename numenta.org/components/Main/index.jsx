// Numenta.org HTM Community website source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2016 Numenta <http://numenta.com>

import React from 'react'

import styles from './index.css'


/**
 *
 */
const Main = ({children}) => (
  <main className={styles.main}>
    {children}
  </main>
)

Main.propTypes = {
  children: React.PropTypes.node.isRequired,
}

export default Main
