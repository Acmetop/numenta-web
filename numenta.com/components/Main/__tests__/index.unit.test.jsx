import React from 'react'
import renderer from 'react-test-renderer'

import Main from '../../Main'


describe('<Main /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <Main />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
