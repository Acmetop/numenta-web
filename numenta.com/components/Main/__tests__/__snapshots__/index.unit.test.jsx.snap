exports[`<Main /> React component unit test suite Renders correctly 1`] = `
<main
  className="main">
  <div>
    Hello
  </div>
</main>
`;
